extensions/common/gcp/gcp-core/src/test/java/org/eclipse/edc/gcp/common/GcpCredentialsTest.java

```diff
@@ -0,0 +1,182 @@
+                mbWKYTu6gTDs39IP1ZqMLXWzVPCCIWCCI3I/iwKBgArrjiqmWQ55xRbdGCvw5RCv
+                cIyuvwZTRYGrkGDIq7HHcFQDTqcxMMQhe0ox9WgQDuu8hAsrVcJv2Ia/14W3xQPT
+                yGAc9EtS112c3ZOYCyuPY4NgoIad9TvpHvpiaLNukbIUAIimoNbyWAUwmkTjTjkC
+                4X+WEhgfmekYzcrk6nfS
+                -----END PRIVATE KEY-----
+                """;
+        var serviceAccountFileInJson = String.format("""
+                {
+                  "type": "service_account",
+                  "project_id": "project-test-1",
+                  "private_key_id": "id1",
+                  "private_key": "%s",
+                  "client_email": "test@project-test-1.iam.gserviceaccount.com",
+                  "client_id": "client_id1",
+                  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+                  "token_uri": "https://oauth2.googleapis.com/token",
+                  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+                  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/project-test-1.iam.gserviceaccount.com"
+                }""", privateKey);
+
+        var tokenInJson = """
+                {"edctype":"dataspaceconnector:gcptoken","token":"18f4155e-4d02-4629-8804-057a3e9277ba","expiration":1665998128000}""";
+
+        serviceAccountFileInB64 = Base64.getEncoder().encodeToString(serviceAccountFileInJson.getBytes());
+        var tokenInB64 = Base64.getEncoder().encodeToString(tokenInJson.getBytes());
+
+        when(vault.resolveSecret(accessTokenKeyName)).thenReturn(tokenInJson);
+        when(vault.resolveSecret(invalidAccessTokenKeyName)).thenReturn("{\"token\":\"" + tokenValue + "\"");
+        when(vault.resolveSecret(serviceAccountKeyName)).thenReturn(serviceAccountFileInJson);
+        when(vault.resolveSecret(serviceAccountKeyName)).thenReturn(serviceAccountFileInJson);
+        when(vault.resolveSecret(invalidServiceAccountKeyName)).thenReturn("{\"type\": \"service_account\" }");
+        gcpCredential = new GcpCredentials(vault, new TypeManager(), mock(Monitor.class));
+    }
+
+    @Test
+    void testResolveGoogleCredentialWhenTokenKeyNameIsProvided() {
+        var gcpServiceAccountCredentialsWithTokenKeyName = new GcpServiceAccountCredentials(accessTokenKeyName, null, null);
+        var gcpCred = gcpCredential.resolveGoogleCredentialsFromDataAddress(gcpServiceAccountCredentialsWithTokenKeyName);
+        assertThat(gcpCred != null);
+    }
+
+    @Test
+    void testResolveGoogleCredentialWhenInvalidTokenIsProvided() {
+        var gcpServiceAccountCredentialsWithInvalidTokenKeyName = new GcpServiceAccountCredentials(invalidAccessTokenKeyName,  null, null);
+        Exception thrown = assertThrows(EdcException.class, () -> gcpCredential.resolveGoogleCredentialsFromDataAddress(gcpServiceAccountCredentialsWithInvalidTokenKeyName));
+        assertThat(thrown.getMessage().contains("valid GcpAccessToken format"));
```

Comment:
I don't think this is the correct way of using `AssertJ` assertion, you need pass only the result object to the `assertThat` method and then add the assertions fluently, in this case like:
```java
assertThat(thrown).message().contains("valid GcpAccessToken format")
```

in this case in fact you could use the `assertThatThrownBy` (because `assertThrows` is not from AssertJ).
same for all the other assertions.

please check the docs for further information: https://assertj.github.io/doc/